### DISCUSSION TITLE
input.http or input.harbor FLAC stream impossible to get metadata

### DISCUSSION THREAD

**Discussion by jduffas** [savonet/liquidsoap#3004] [User Support]
Hello,
I'm using version 2.1.4 
I'm experiencing a problem that I didn't have with version 1.x
it's now impossible to get metadata from FLAC stream.

my code is :
```
source_Flac = input.http("http://localhost:8000/source_TEST")

source = mksafe(source_Flac)

output.icecast(%fdkaac(aot="mpeg4_aac_lc", bitrate=128), host = "localhost", port = 8000, password = "xxx", mount = "singsingaac128TEST", name="Radio Sing Sing aac 128 Kb/s", description="Un nouveau souffle pour vos oreilles", url="http://sing-sing-bis.org", genre="eclectic", source)
```

Same problem if I change the input way, with harbor :

`source_Flac = input.harbor("Source_TEST",port=8080,password="xxx")`

I installed the Ubuntu : liquidsoap_2.1.4-ubuntu-focal-1_amd64.deb
idem with the 2.2.0+git@2a3bf62e1 version.
the icecast is the 2.5 Beta 3 (I got the same result with 2.4.4 version)

any idea?

**Comment by jduffas** [comment]
I found the solution, I had to add this line at the beginning of the script 
`settings.decoder.mime_types.ffmpeg.set(["FLAC"])`

**Reply by toots** [reply]
I see thanks for reporting. This looks like a bug with the ffmpeg code that is specific to ogg/flac. ogg/vorbis and ogg/opus seem to work just fine.

If I understand well your solution, you are probably using `input.harbor` with it.  This settings probably simply disable the ffmpeg stream decoder, which has the highest priority and, so, I imagine your stream is being decoded by the native ogg decoder.

With `input.http`, we are limited to the ffmpeg decoder only, so the bug remains. I'll see if I can isolate it and either report or send a patch upstream.

**Comment by jduffas** [comment]
Yes, I use` input.harbor` with it.
Thank you for your answer.
